search/elastic.py

```diff
@@ -579,35 +558,34 @@ def search(self,
         if elastic_filters:
             filter_segment = {
                 "bool": {
-                    "must": elastic_filters
+                    "should": elastic_filters
                 }
             }
             query = {
-                "filtered": {
-                    "query": query_segment,
+                "bool": {
+                    "must": query_segment,
                     "filter": filter_segment,
                 }
             }
 
-        body = {"query": query}
-        if facet_terms:
-            facet_query = _process_facet_terms(facet_terms)
-            if facet_query:
-                body["facets"] = facet_query
+        if exclude_dictionary:
+            if query.get("bool"):
+                query["bool"]["must_not"] = _process_exclude_dictionary(exclude_dictionary)
+            else:
+                query = {
+                    "bool": {
+                        "must_not": _process_exclude_dictionary(exclude_dictionary)
+                    }
+                }
+
+        body = {"query": query, }
+        if agg_terms:
+            body["aggs"] = _process_aggregation_terms(agg_terms)
 
         try:
-            es_response = self._es.search(
-                index=self.index_name,
-                body=body,
-                **kwargs
-            )
+            es_response = self._es.search(index=self.index_name, body=body, **kwargs)
         except exceptions.ElasticsearchException as ex:
-            message = str(ex)
-            if 'QueryParsingException' in message:
-                log.exception("Malformed search query: %s", message)  # lint-amnesty, pylint: disable=unicode-format-string
-                raise QueryParseError('Malformed search query.')
-            # log information and re-raise
-            log.exception("error while searching index - %s", str(message))  # lint-amnesty, pylint: disable=unicode-format-string
+            log.exception("error while searching index - %s", str(ex))
```

Comment:
str, here the same. I am not 100% sure but feel that it is to much.

Reply:
What do you mean about to much?

Reply:
I forgot )
```
log.exception("error while searching index - %r", ex)
```